mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_network_policy_test.go

```diff
@@ -0,0 +1,75 @@
+package vmwareengine_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccVmwareengineNetworkPolicy_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"region":        envvar.GetTestRegionFromEnv(),
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckVmwareengineNetworkPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVmwareengineNetworkPolicy_config(context, false, false),
+			},
+			{
+				ResourceName:            "google_vmwareengine_network_policy.vmw-engine-network-policy",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name"},
+			},
+			{
+				Config: testAccVmwareengineNetworkPolicy_config(context, true, true),
+			},
+			{
+				ResourceName:            "google_vmwareengine_network_policy.vmw-engine-network-policy",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name"},
+			},
+		},
+	})
+}
+
+func testAccVmwareengineNetworkPolicy_config(context map[string]interface{}, internetAccess bool, externalIp bool) string {
```

Comment:
We should be checking all updateable fields here, so we'd expect to see changes to `description`, `edge_services_cidr`, and `vmware_engine_network`, based on the current implementation. On a similar token, it would be best if one of those configs also omitted all optional fields, to validate that they are not required. Using 2 separate config templates might make that easier to test.

Reply:
Thanks for the feedback Ryan, I've modified the update test to include changes in all mutable fields. I have made also added two seperate tests in examples for _basic and _full to validate the creation of resource with and without all optional fields.